service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/ais/AccountSpiImpl.java

```diff
@@ -1,59 +1,136 @@
 package de.adorsys.psd2.sandbox.xs2a.service.ais;
 
+import de.adorsys.psd2.sandbox.portal.testdata.TestDataService;
+import de.adorsys.psd2.sandbox.portal.testdata.domain.Account;
+import de.adorsys.psd2.sandbox.portal.testdata.domain.Balance;
 import de.adorsys.psd2.xs2a.core.consent.AspspConsentData;
 import de.adorsys.psd2.xs2a.spi.domain.account.SpiAccountBalance;
 import de.adorsys.psd2.xs2a.spi.domain.account.SpiAccountConsent;
 import de.adorsys.psd2.xs2a.spi.domain.account.SpiAccountDetails;
 import de.adorsys.psd2.xs2a.spi.domain.account.SpiAccountReference;
+import de.adorsys.psd2.xs2a.spi.domain.account.SpiBalanceType;
 import de.adorsys.psd2.xs2a.spi.domain.account.SpiTransaction;
 import de.adorsys.psd2.xs2a.spi.domain.account.SpiTransactionReport;
+import de.adorsys.psd2.xs2a.spi.domain.common.SpiAmount;
 import de.adorsys.psd2.xs2a.spi.domain.response.SpiResponse;
 import de.adorsys.psd2.xs2a.spi.service.AccountSpi;
 import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 public class AccountSpiImpl implements AccountSpi {
 
+  TestDataService testDataService;
+
+  @Autowired
+  public AccountSpiImpl() {
+    testDataService = new TestDataService();
+  }
+
   @Override
   public SpiResponse<List<SpiAccountDetails>> requestAccountList(boolean b,
       @NotNull SpiAccountConsent spiAccountConsent,
       @NotNull AspspConsentData aspspConsentData) {
-    return null;
+
+    List<SpiAccountReference> accountList = spiAccountConsent.getAccess().getAccounts();
+    List<String> ibans = new ArrayList<>();
+
+    for (SpiAccountReference account : accountList) {
+      ibans.add(account.getIban());
+    }
+
+    String psuId = testDataService.getPsuByIban(ibans.get(0));
+    List<Account> accounts = testDataService.getRequestedAccounts(psuId, ibans);
+    List<SpiAccountDetails> spiAccountDetails = mapAccountToSpiAccount(accounts);
+
+    return SpiResponse.<List<SpiAccountDetails>>builder()
+        .aspspConsentData(aspspConsentData)
+        .payload(spiAccountDetails)
+        .success();
+  }
+
+  private List<SpiAccountDetails> mapAccountToSpiAccount(List<Account> accountList) {
+    List<SpiAccountDetails> spiAccountDetails = new ArrayList<>();
+
+    accountList.stream().forEach(account -> spiAccountDetails.add(
```

Comment:
Use `accountList.stream().map(acc -> this::mapAccount)` or something like that to get from `List<Account>` to `List<SpiAccountDetails>`.

Reply:
Done